contracts/SyntheticAssetV1.sol

```diff
@@ -51,53 +50,63 @@ contract SyntheticAssetV1 is ERC20 {
+        require(_amount > 0, "Amount should be greator than zero");
+
         _mint(_userAddress, _amount);
-        userToSynthAssetEligibleToBurn[_userAddress] += _amount;
+        addressToAmountEligibleToBurn[_userAddress] += _amount;
+        emit Mint(_userAddress, _amount);
     }
 
     /**
      * @notice Burn synthetic assets.
      * @param _userAddress the address of the user.
      * @param _amount the amount of assets that gets burnt.
      */
-     //slither-disable-next-line naming-convention
+    //slither-disable-next-line naming-convention
     function burn(address _userAddress, uint _amount) external onlyHub {
-        require(_amount <= userToSynthAssetEligibleToBurn[_userAddress], "The user has not enough assets");
-        userToSynthAssetEligibleToBurn[_userAddress] -= _amount;
+        require(_userAddress != address(0), "The address parameter cannot be null");
+        require(_amount > 0, "Amount should be greator than zero");
+        require(_amount <= addressToAmountEligibleToBurn[_userAddress], "The user has not enough assets");
+
         _burn(_userAddress, _amount);
+        addressToAmountEligibleToBurn[_userAddress] -= _amount;
+        emit Burn(_userAddress, _amount);
     }
 
     /**************************************** Getters ****************************************/
 
     /**
      * @notice Return the amount of synthetic asset a user is elligible to burn.
      * @param _userAddress the address of the user.
-     * @return _ the amount of synthetic asset elligible to be burnt.
+     * @return _ the amount of synthetic asset the user is elligible to burn.
      */
-     //slither-disable-next-line naming-convention
-    function getUserToSynthAssetEligibleToBurn(address _userAddress) external view returns (uint256) {
-        return userToSynthAssetEligibleToBurn[_userAddress];
+    //slither-disable-next-line naming-convention
+    function getAmountEligibleToBurn(address _userAddress) external view returns (uint256) {
+        require(_userAddress != address(0), "The address parameter cannot be null");
+        return addressToAmountEligibleToBurn[_userAddress];
     }
 
     /**************************************** Setters ****************************************/
 
     /**
      * @notice Update the amount of synthetic asset a user is elligible to burn.
      * @param _userAddress the address of the user.
-     * @param _amount the amount the user is elligible to burn.
+     * @param _amount the amount of synthetic asset the user is elligible to burn.
      */
-     //slither-disable-next-line naming-convention
-    function setUserToSynthAssetEligibleToBurn(address _userAddress, uint256 _amount) external onlyHub {
-        userToSynthAssetEligibleToBurn[_userAddress] = _amount;
+    //slither-disable-next-line naming-convention
+    function setAmountEligibleToBurn(address _userAddress, uint256 _amount) external onlyHub {
```

Comment:
add to the doc that this function is used to liquidate users

Reply:
Yep, I added a dev comment to the method's documentation.